src/shared/ui/buttons.ts

```diff
@@ -3,25 +3,59 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-import { QuickInputButton, Uri } from 'vscode'
+import * as vscode from 'vscode'
+import * as nls from 'vscode-nls'
 import { ext } from '../extensionGlobals'
+import { WizardControl } from '../wizards/wizard'
+
+const localize = nls.loadMessageBundle()
+const HELP_TOOLTIP = localize('AWS.command.help', 'View Toolkit Documentation')
+
+type WizardButton<T> = QuickInputButton<T | WizardControl> | QuickInputButton<void>
```

Comment:
Because `QuickInputButton<T | WizardControl | void>[]` is not the same as `(QuickInputButton<T | WizardControl> | QuickInputButton<void>)[]`

Array types are not distributive over the generic union-ed type. The latter describes an array of buttons where _every_ button could return `T | WizardControl | void`, the former describes an array that may contain buttons that return `T | WizardControl` _or_ buttons that return `void`.

Reply:
I understand this, but I'm struggling to think of where the distinction becomes important? No matter what happens in that array case, you'll have to handle a button returning potentially all three types.

Reply:
It's not important for the implementation, it's important for the compiler to infer types.